FirebaseSessions/Sources/Settings/SessionsSettings.swift

```diff
@@ -0,0 +1,86 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Class that manages the configs related to the settings library
+class SessionsSettings {
+  private let appInfo: ApplicationInfoProtocol
+  private let installations: InstallationsProtocol
+  private let sdkDefaults: SDKDefaultSettings
+  private let localOverrides: LocalOverrideSettings
+  private let remoteSettings: RemoteSettings
+
+  convenience init(appInfo: ApplicationInfoProtocol, installations: InstallationsProtocol) {
+    self.init(appInfo: appInfo,
+              installations: installations,
+              sdkDefaults: SDKDefaultSettings(),
+              localOverrides: LocalOverrideSettings(),
+              remoteSettings: RemoteSettings(appInfo: appInfo,
+                                             downloader: SettingsDownloader(appInfo: appInfo,
+                                                                            installations: installations)))
+  }
+
+  init(appInfo: ApplicationInfoProtocol,
+       installations: InstallationsProtocol,
+       sdkDefaults: SDKDefaultSettings,
+       localOverrides: LocalOverrideSettings,
+       remoteSettings: RemoteSettings) {
+    self.appInfo = appInfo
+    self.installations = installations
+    self.sdkDefaults = sdkDefaults
+    self.localOverrides = localOverrides
+    self.remoteSettings = remoteSettings
+  }
+
+  var sessionsEnabled: Bool {
+    // Order of precendence LocalOverrides > Remote Settings > SDK Defaults
+    if !localOverrides.isSettingsStale(), let sessionEnabled = localOverrides.sessionsEnabled {
+      return sessionEnabled
+    } else if !remoteSettings.isSettingsStale(),
+              let sessionEnabled = remoteSettings.sessionsEnabled {
+      return sessionEnabled
+    }
+    return sdkDefaults.sessionsEnabled!
+  }
+
+  var sessionTimeout: TimeInterval {
+    // Order of precendence LocalOverrides > Remote Settings > SDK Defaults
+    if !localOverrides.isSettingsStale(), let sessionTimeout = localOverrides.sessionTimeout {
+      return sessionTimeout
+    } else if !remoteSettings.isSettingsStale(),
+              let sessionTimeout = remoteSettings.sessionTimeout {
+      return sessionTimeout
+    }
+    return sdkDefaults.sessionTimeout!
+  }
+
+  var samplingRate: Double {
+    // Order of precendence LocalOverrides > Remote Settings > SDK Defaults
+    if !localOverrides.isSettingsStale(), let samplingRate = localOverrides.samplingRate {
+      return samplingRate
+    } else if !remoteSettings.isSettingsStale(), let samplingRate = remoteSettings.samplingRate {
```

Comment:
I wonder if we can remove the `isSettingsStale` call altogether and instead just rely on the `if let ... = ...` being nil for each value.

Reply:
I like this. As a config settings fetcher, we don't have to worry about the staleness of configs.